packages/aws-cdk-lib/aws-servicecatalog/lib/private/product-stack-synthesizer.ts

```diff
@@ -10,45 +9,42 @@ import { ProductStack } from '../product-stack';
  * Interoperates with the StackSynthesizer of the parent stack.
  */
 export class ProductStackSynthesizer extends cdk.StackSynthesizer {
-  private readonly assetBucket?: IBucket;
   private bucketDeployment?: BucketDeployment;
+  private parentAssetBucket?: IBucket;
 
-  constructor(assetBucket?: IBucket) {
+  constructor(private readonly parentDeployment: cdk.IStackSynthesizer, private readonly assetBucket?: IBucket) {
```

Comment:
```suggestion
  constructor(private readonly parentStack: cdk.Stack, private readonly assetBucket?: IBucket) {
```
I know this is only an internal class, but I wonder if passing the parent stack is a bit cleaner as an API. Passing a synthesizer feels like we are implying that any synthesizer could be passed in here.

Reply:
I've updated the code to pass the `parentStack` as part of the newly added `ProductStackSynthesizerProps`